.gitignore

```diff
@@ -7,12 +23,61 @@ build/
 # Local configuration file (sdk path, etc)
 local.properties
 
-.idea/*
-!.idea/codeStyles/
-!.idea/codeStyleSettings.xml
-!.idea/inspectionProfiles/
-.DS_Store
+# Proguard folder generated by Eclipse
+proguard/
 
-.DS_Store
-/captures
+# Log Files
+*.log
+
+# Android Studio Navigation editor temp files
+.navigation/
+
+# Android Studio captures folder
+captures/
+
+# IntelliJ
+*.iml
+.idea/workspace.xml
+.idea/tasks.xml
+.idea/gradle.xml
+.idea/assetWizardSettings.xml
+.idea/dictionaries
+.idea/libraries
+# Android Studio 3 in .gitignore file.
+.idea/caches
+.idea/modules.xml
+# Comment next line if keeping position of elements in Navigation Editor is relevant for you
+.idea/navEditor.xml
```

Comment:
I'm not sure about this approach, isn't it better to ignore whole .idea folder and explicitly state exceptions?